src/stores/RedisStore.ts

```diff
@@ -0,0 +1,33 @@
+import { connect } from "../../deps.ts";
+import type { Ratelimit } from "../types/types.d.ts";
+import { Store } from "./AbstractStore.ts";
+
+export class RedisStore extends Store {
+    private readonly store;
+
+    constructor(hostname?: string, port?: number) {
+        super();
+        this.store = connect({
+            hostname: hostname ?? "localhost",
+            port
+        })
+    }
+
+    public async get(ip: string) {
+        const data = await (await this.store).get(ip)
```

Comment:
I would prefer you have any statements which require a callback (`await`) to be their own variable to avoid promise stacking and make for cleaner and understandable code.